ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp

```diff
@@ -1,25 +1,112 @@
 #include "PFmain.h"
 
+#include "Platforms/platforms_mandatory.h"
+
+#include <unistd.h>  //getcwd, usleep
+
 namespace enigma {
 
 bool game_isending = false;
 int game_return = 0;
+int pausedSteps = 0;
+int current_room_speed;
+std::string *parameters;
+int parameterc;
+int frames_count = 0;
+unsigned long current_time_mcs = 0;
 
 long clamp(long value, long min, long max) {
   if (value < min) return min;
   if (value > max) return max;
   return value;
 }
 
+int gameWait() {
+  if (enigma_user::os_is_paused()) {
+    if (pausedSteps < 1) {
+      pausedSteps += 1;
+    } else {
+      usleep(100000);
+      return -1;
+    }
+  }
+
+  frames_count++;
+
+  return 0;
+}
+
+void set_room_speed(int rs) { current_room_speed = rs; }
+
+unsigned long get_timer() {  // microseconds since the start of the game
+  return current_time_mcs;
+}
+
+void set_program_args(int argc, char** argv) {
+  parameters = new std::string[argc];
+  parameterc = argc;
+  for (int i = 0; i < argc; i++) parameters[i] = argv[i];
+}
+
+int main(int argc, char** argv, void* windowHandle) {
+  // Set the working_directory
+  set_working_directory();
+
+  // Copy our parameters
+  set_program_args(argc, argv);
+
+  initInput();
+
+  if (!initGameWindow()) {
+    printf("Failed to create game window\n");
+    return -4;
+  }
+
+  EnableDrawing(windowHandle);
```

Comment:
Perhaps this is another good use case of extension hooks. Enabling window drawing could just be a routine that each bridge adds. It does sound like a trick to integrate that into the configuration, though; due to the nature of being the system's initializer, it wouldn't have an auxiliary initializer to register the hook, so these would have to be listed in the eyaml (or else use a nasty hack to append to a hook list). I can explain if you're interested.